Content.Shared/Research/Systems/BlueprintSystem.cs

```diff
@@ -0,0 +1,108 @@
+using Content.Shared.IdentityManagement;
+using Content.Shared.Interaction;
+using Content.Shared.Lathe;
+using Content.Shared.Popups;
+using Content.Shared.Research.Components;
+using Content.Shared.Research.Prototypes;
+using Content.Shared.Whitelist;
+using Robust.Shared.Containers;
+using Robust.Shared.Prototypes;
+
+namespace Content.Shared.Research.Systems;
+
+public sealed class BlueprintSystem : EntitySystem
+{
+    [Dependency] private readonly SharedContainerSystem _container = default!;
+    [Dependency] private readonly EntityWhitelistSystem _entityWhitelist = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<BlueprintReceiverComponent, ComponentStartup>(OnStartup);
+        SubscribeLocalEvent<BlueprintReceiverComponent, AfterInteractUsingEvent>(OnAfterInteract);
+        SubscribeLocalEvent<BlueprintReceiverComponent, LatheGetRecipesEvent>(OnGetRecipes);
+    }
+
+    private void OnStartup(Entity<BlueprintReceiverComponent> ent, ref ComponentStartup args)
+    {
+        _container.EnsureContainer<Container>(ent, ent.Comp.ContainerId);
```

Comment:
this is typically done just so that components whose logic necessarily relies on containers don't experience breakages or crashes when added onto entities via VV. It's not really all that necessary (not having the container on the prototype will cause test fails) but its just a safety to prevent crashes and enable things to be done outside of the prototype.